File: solutions/2639-separate-the-digits-in-an-array/separate-the-digits-in-an-array.go
func separateDigits(nums []int) []int {
	res := []int{}
	for i := len(nums) - 1; i >= 0; i-- {
		x := nums[i]
		for x > 0 {
			res = append(res, x%10)
			x /= 10
		}
	}

	for i, j := 0, len(res)-1; i < j; i, j = i+1, j-1 {
		res[i], res[j] = res[j], res[i]
	}

	return res
}

File: solutions/2639-separate-the-digits-in-an-array/separate-the-digits-in-an-array.cs
class Solution {
    public int[] SeparateDigits(int[] nums) {
        List<int> res = new List<int>();
        for (int i = nums.Length - 1; i >= 0; i--) {
            int x = nums[i];
            while (x > 0) {
                res.Add(x % 10);
                x /= 10;
            }
        }
        res.Reverse();
        return res.ToArray();
    }
}